src/utils/SyncProvider/SyncDroppedTransaction.tsx

```diff
@@ -0,0 +1,202 @@
+    // same, we can assume it was a replacement. Best we can do since we don't have the nonce
+    const replacementTransactions = accountTransactionHistory.filter(
+      (historicTransaction) =>
+        historicTransaction.input === searchingTransaction.input &&
+        parseInt(historicTransaction.timeStamp, 10) > (searchingTransaction?.timestamp ?? 0),
+    )
+
+    if (replacementTransactions.length > 1) {
+      console.error('Ambiguous transaction replacement')
+      return
+    }
+
+    if (replacementTransactions.length === 1) {
+      const _replacementTransaction = replacementTransactions[0]
+      store.setReplacedTransaction(
+        address,
+        chainId,
+        _replacementTransaction.input,
+        _replacementTransaction,
+      )
+      return
+    }
+
+    // If for some reason the transaction was not found, was not a replacement etc, try to find it again.
+    const result = await provider.getTransaction(searchingTransaction.hash)
+    if (result) {
+      store.foundTransaction(address, chainId, searchingTransaction.hash, result.nonce)
+      return
+    }
+    store.updateRetries(address, chainId, searchingTransaction.hash)
+  }
+
+  for (const pendingTransaction of pendingTransactions) {
+    const currentNonce = await provider.getTransactionCount(address)
+
+    if (currentNonce > (pendingTransaction?.nonce ?? -1)) {
+      // Transaction either got replaced or has been cancelled
+
+      // Find tranasaction in user's history based on nonce
+      // Get matching nonce from history
+      const matchingNonceTransaction = accountTransactionHistory.find((tx: any) => {
+        return parseInt(tx.nonce, 10) === pendingTransaction.nonce
+      })
+
+      // See if matching nonce transaction is a replacement
+      if (
+        matchingNonceTransaction &&
+        matchingNonceTransaction?.input === pendingTransaction?.input
+      ) {
+        store.setReplacedTransactionByNonce(
+          address,
+          chainId,
+          pendingTransaction?.input ?? ``,
+          matchingNonceTransaction,
+        )
+        return
+      }
+
+      // If there is a gap between account nonce and latest nonce in history
+      // it is possible that etherscan history is behind
+      if (currentNonce - parseInt(accountTransactionHistory[0].nonce, 10) > 1) {
+        // Wait for etherscan history to update
+        return
+      }
+
+      // If the transaction was not replaced then it is a failed transaction
+      store.setFailedTransaction(address, chainId, pendingTransaction.hash)
+    }
+
+    // If the transaction has not been cancelled or replaced, it may have been dropped
+    const result = await provider.getTransaction(pendingTransaction.hash)
+    if (!result) {
+      // If a pending transaction is not found, it has been dropped
+      store.setFailedTransaction(address, chainId, pendingTransaction.hash)
+      return
+    }
+  }
```

Comment:
since these for...of loops contain promises these should probs be wrapped in `Promise.all`

Reply:
I'll give it a go in a follow up